### PR TITLE
battstat: Rewrite battstat_applet.ui from scratch

### PR DESCRIPTION
**Pull request by rbuj**
The resource files are loaded from memory instead of disk, removing the read latency and the number of small block disk accesses. It uses a dialog template for a preference dialog, updating and simplifying the code. Doing so in this way reduces the applet complexity as you can see the number of attributes in ProgressData has been reduced. It fixes an issue with an enumeration, in which the first item is not initialized to zero.

Test: Open preferences dialog

### CODE REVIEW

File: battstat/battstat-preferences.c [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+#include <unistd.h>
+
+#include <gtk/gtk.h>
+
+#include <gio/gio.h>
+
+#include "battstat-preferences.h"
+
+enum {
+    PROP_0,
```
Comment (raveit65) [inline battstat/battstat-preferences.c:50]: Personal i prefer 4 spaces at the beginning but the whole file use 2 spaces now.
So you should use 2 spaces here.

**Comment by raveit65** [comment]
The GtkSpinButton is a bit large. For me it looks better when using `haglin` `GTK_ALIGN_START`.
batstatt-pref-new
batstatt-pref-new-better

File: battstat/battstat-preferences.c (hunk 1 above)
Comment (rbuj) [reply battstat/battstat-preferences.c]: done

**Comment by rbuj** [comment]
@raveit65 done

**Comment by raveit65** [comment]
On my notebook the settings for remaining time or percent are always gey out. (not tested with latest update)
battstat-notebook
And time or percent values aren`t displayed in panel

**Comment by rbuj** [comment]
@raveit65 done

**Comment by raveit65** [comment]
Thanks, this is fixed, but only percent values are displayed in panel, time values are failing.
And i noticed that the applet won't displayed in panel when you add it the first time.
After resizing the panel i see 2 applets , deja vu...
And i think the icon size is sometimes wrong for the panel size.

**Comment by raveit65** [comment]
Btw. I am not sure if displaying the time in panel was broken before. Honestly i never use this applet normally.

**Comment by sc0w** [comment]
which are the benefits of this change?

why changing a lot of code if works?

**Comment by rbuj** [comment]
> Thanks, this is fixed, but only percent values are displayed in panel, time values are failing.

done

> And i noticed that the applet won't displayed in panel when you add it the first time.
> After resizing the panel i see 2 applets , deja vu...
> And i think the icon size is sometimes wrong for the panel size.

It's not related to this PR, open a ticket to report about this issue.

**Comment by rbuj** [comment]
> which are the benefits of this change?
> 
> why changing a lot of code if works?

where is the review, I can't find it in the text you wrote ....

**Comment by raveit65** [comment]
> which are the benefits of this change?
> why changing a lot of code if works?

When i read changes i see:
- replacing self backed function with gtk ones.
- loading menu xml and ui files as resource
- replacing deprecated gtk code
Isn't this OK?

**Comment by raveit65** [comment]
> It's not related to this PR, open a ticket to report about this issue.

OK, i will do after final testing........tomorrow :)

**Comment by sc0w** [comment]
@rbuj omg, I can't do questions?

@raveit65 

> When i read changes i see:
> 
>     replacing self backed function with gtk ones.
>     loading menu xml and ui files as resource
>     replacing deprecated gtk code

Thanks! :)

**Comment by sc0w** [comment]
I mean, the commit message can be more descriptive

**Comment by rbuj** [comment]
@sc0w similar commit https://github.com/GNOME/gnome-panel/commit/77a281ab38747315fd3f6a2802b673cd0d065b7a

**Comment by raveit65** [comment]
I agree with writing more informations in commit body. This is a huge code change and i can't review that all for myself.
Another reviewer i definitely needed.
But guys, seeing in every PR your personal conflicts is complete boring.
Barca : Real = 0:0

**Comment by rbuj** [comment]
@raveit65  He recently asked for the same in a PR made by @mbkma https://github.com/mate-desktop/pluma/pull/563#issuecomment-669655899

> I don't understand the benefits of this change, it improves performance?

About if @sc0w have a conflict with me. It's the first news I have. I think he behaves the same with everyone, I have not noticed any different behavior with me.

**Comment by sc0w** [comment]
@rbuj I have no problems with you, I am just reviewing PRs

@rbuj 

> He recently asked for the same in a PR made by @mbkma mate-desktop/pluma#563 (comment)

That PR isn't related with this, I don't understand you, the question in that PR is for the author, not for you, if you like to say me something related with that, you can do it there.

@rbuj 

> similar commit GNOME/gnome-panel@77a281a

This is not gnome

**Comment by rbuj** [comment]
> That PR isn't related with this, I don't understand you, the question in that PR is for the author, not for you, if you like to say me something related with that, you can do it there.

It's because that PR had a description and this one do not, and the result is the same question in a few hours of difference.

**Comment by sc0w** [comment]
@rbuj 

I see no description of the changes here, that's all, and I will tell you again if there is no description in other PRs, we need to understand why the changes.

In other PR I just do the question why calling 30 times `g_settings_new` is better than 12 times but this is not the place to talk about other PR

**Comment by rbuj** [comment]
I'm writing it :)

**Comment by rbuj** [comment]
The resource files are loaded from memory instead of disk, removing the read latency and the number of small block disk accesses. It uses a dialog template for a preference dialog, updating and simplifying the code. Doing so in this way reduces the applet complexity as you can see the number of attributes in ProgressData is reduced

**Comment by sc0w** [comment]
@rbuj Thanks, can you write that description it in the next lines of commit message please? :)

**Comment by rbuj** [comment]
done, something else?

**Comment by sc0w** [comment]
you didn't change commit message

**Comment by rbuj** [comment]
I don't think it's necessary. However, what title do you propose?

**Comment by sc0w** [comment]
more lines describing the changes, it is hard to understand?

you can't describe the changes in commit?

the author are you, not me

**Comment by rbuj** [comment]
Now I understand you. You are asking to add the description in the commit body. We talked about this a long time ago, and we made the decision not to add it, if not strictly necessary, in order to keep polished logs. However you can add a description in the PR on github, as well as other useful information to validate or review the request.

@raveit65 do you remember talking about this in another PR?

**Comment by rbuj** [comment]
@sc0w I understand your difficulties in speaking in English, but review the ways in which you express yourself.

**Comment by raveit65** [comment]
@rbuj 
Yes i remember. But you wrote a whole book in this PR :) . For this reason i asked you to write full description in PR and a minimal description in commit body.
I said my opinion already in this PR.
https://github.com/mate-desktop/mate-applets/pull/497#issuecomment-670196881
> I agree with writing more informations in commit body.

Please....to finish this noise.

**Comment by rbuj** [comment]
done

**Comment by raveit65** [comment]
Showing the remaining time in panel is working now, but i think calculating the time is wrong with both applets, this one and the one from m-p-m.
After unplug both applets are showing 99% and 95 hours 0_o
Than 98 % and 3.6 hours.
Next 97% and 2.5 hours
Now 94 and 3 hours and 3 min

Anyway, not related to PR because both applets showing the same.

> And i noticed that the applet won't displayed in panel when you add it the first time.
> After resizing the panel i see 2 applets , deja vu...
> And i think the icon size is sometimes wrong for the panel size.

Surprise, at the moment it's not reproducible any more.

**Comment by raveit65** [comment]
I debian build log i found new warnings which i don't see in old files from master.
https://travis-ci.org/github/mate-desktop/mate-applets/jobs/715809383#L6998
```
battstat-preferences.c:74:26: warning: unused parameter 'combo_ptr' [-Wunused-parameter]

combo_ptr_cb (GtkWidget *combo_ptr,
```
https://travis-ci.org/github/mate-desktop/mate-applets/jobs/715809383#L7055
```
battstat-preferences.c:90:25: warning: unused parameter 'spin_ptr' [-Wunused-parameter]

spin_ptr_cb (GtkWidget *spin_ptr,
```
https://travis-ci.org/github/mate-desktop/mate-applets/jobs/715809383#L7063
```
battstat-preferences.c:222:50: warning: unused parameter 'pspec' [-Wunused-parameter]

                                   GParamSpec   *pspec)

                                                 ^                       ^

battstat_applet.c:1154:battstat-preferences.c:239:48: warning: unused parameter 'pspec' [-Wunused-parameter]

                                   GParamSpec *pspec)
```

**Comment by rbuj** [comment]
@raveit65 done

**Comment by raveit65** [comment]
Offtopic, i did run in icon problem again. i think it's related to the icon-set for loading the battery.

**Comment by rbuj** [comment]
> Offtopic, i did run in icon problem again. I think it's related to the icon-set for loading the battery.

@raveit65 can you test if #525 fixes the error? if it doesn't work I'll delete the PR.

**Comment by rbuj** [comment]
> I only notice warnings in the logs already presents in master

Yes, we removed 6 warning from all build logs in last 48 hours:
```
$ diff 2020-08-05.txt 2020-08-07.txt 
16c16
< -Wenum-conversion 27
---
> -Wenum-conversion 26
33c33
< -Wtype-limits 15
---
> -Wtype-limits 13
36,37c36,37
< -Wunused-macros 789
< -Wunused-parameter 6870
---
> -Wunused-macros 787
> -Wunused-parameter 6869
39c39
< Total: 17223/30 (num/types)
---
> Total: 17217/30 (num/types)
```

**Comment by sc0w** [comment]
and this one:
```

In file included from battstat-preferences.c:32:
/usr/include/x86_64-linux-gnu/sys/sysctl.h:21:2: warning: "The <sys/sysctl.h> header is deprecated and will be removed." [-W#warnings]
#warning "The <sys/sysctl.h> header is deprecated and will be removed."
 ^
```

**Comment by rbuj** [comment]
@sc0w ???

**Comment by rbuj** [comment]
-Wcpp was already present in the build logs, It was not introduced in the last 48 hours. 
```
mate-applets.make.log:/usr/include/sys/sysctl.h:21:2: warning: #warning "The <sys/sysctl.h> header is deprecated and will be removed." [-Wcpp]
mate-applets.make.log:   21 | #warning "The <sys/sysctl.h> header is deprecated and will be removed."
```

**Comment by sc0w** [comment]
yes yes, I know, please dont panic :)

**Pull request merged**